docs/RUP_Architecture.md

```diff
@@ -95,35 +130,160 @@ The frontend is in charge of providing a human-useable interface that provides f
+
+3. Data Storage Process: This process is responsible for storing cashflow data in the Django backend. It receives input from the web worker process and stores the data in the database. The mode of communication between this process and other processes is through message passing (SQL-based queries and replies).
+
+4. Authentication and Authorization Process: This process is responsible for authenticating users and authorizing their access to the app's functionality. It receives input from the user interface process and checks the user's credentials against the Django backend's authentication system. If the user is authorized, this process sends a message back to the user interface process allowing them access to the app's functionality by issuing a JWT. The mode of communication between this process and other processes is through message passing (IPC as well as HTTP).
+
+5. Backend Integration Process: This process is responsible for integrating the Android app with the Django backend. It is capable of managing several lightweight processes (worker threads) that send the requests asynchronously. It receives input from the user interface, and sends it to the appropriate backend API endpoint for processing. Once response data is received, it sends it to the appropriate process for display or further processing. The mode of communication between this process and other processes is through message passing using IPC as well as HTTP for the external API calls.
+
+In summary, the main mode of communication between processes in this system is message passing. The user interface process, cashflow management process, data storage process, authentication and authorization process, and backend integration process all communicate with each other through message passing (whether it is HTTP, IPC or SQL-based).
 
 ### Deployment View
 
-[This section describes one or more physical network (hardware) configurations on which the software is deployed and run. It is a view of the Deployment Model. At a minimum for each configuration it should indicate the physical nodes (computers, CPUs) that execute the software, and their interconnections (bus, LAN, point-to-point, and so on.) Also include a mapping of the processes of the Process View onto the physical nodes.]
+Because the software is not meant to be deployed for wide-range usage, it is only deployed for development as well as presentation purposes.
+
+There is no need to have multiple pieces of hardware.
+However, the running Android app needs to be able to connect to the backend.
+This can be achieved by layer 2 networking.
+
+From a process based view, all Android app related processes run on the (emulated) mobile phone. All other processes run inside a dockerized environment.
+
+In case that an emulated mobile phone is used, and also the backend is running on the same physical machine, it is recommended to have at least four CPU cores.
```

Comment:
by common sense? XD

Reply:
What about this?

```suggestion
In case that an emulated mobile phone is used, and also the backend is running on the same physical machine, we recommend to have at least four CPU cores.
```

In that way it does not sound like somewhere is written down to have atleast four CPU cores. It is just a recommendation by FiBo